components/logger.js

```diff
@@ -4,6 +4,8 @@ import fancyNames from '@/lib/fancy-names.json'
 import { gql, useMutation, useQuery } from '@apollo/client'
 import { WALLET_LOGS } from '@/fragments/wallet'
 import { getWalletBy } from '@/lib/constants'
+// TODO: why can't I import this without errors?
+// import { getWalletByName } from './wallet'
```

Comment:
This import breaks the build with this error:

```
Module build failed: UnhandledSchemeError: Reading from "node:crypto" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
 ⨯ node:crypto
```

This is somehow related to the import of `@/components/wallets/lnbits` in `@/components/wallet`. If I remove the `await import`, it works:

```diff
diff --git a/components/wallet/index.js b/components/wallet/index.js
index fbddb07f..76ac8e18 100644
--- a/components/wallet/index.js
+++ b/components/wallet/index.js
@@ -6,7 +6,7 @@ import { SSR } from '@/lib/constants'

 // wallet definitions
 export const WALLET_DEFS = [
-  await import('@/components/wallet/lnbits')
+  // await import('@/components/wallet/lnbits')
 ]

 export const Status = {
```

But it also doesn't work if I use a "normal import":

```diff
diff --git a/components/wallet/index.js b/components/wallet/index.js
index fbddb07f..9fcb6e21 100644
--- a/components/wallet/index.js
+++ b/components/wallet/index.js
@@ -3,10 +3,11 @@ import { useMe } from '@/components/me'
 import useLocalState from '@/components/use-local-state'
 import { useWalletLogger } from '@/components/logger'
 import { SSR } from '@/lib/constants'
+import lnbits from '@/components/wallet/lnbits'

 // wallet definitions
 export const WALLET_DEFS = [
-  await import('@/components/wallet/lnbits')
+  lnbits
 ]

 export const Status = {
```

fails with same error.

So I realized that `import { bolt11Tags } from '@/lib/bolt11'` in components/wallets/lnbits.js import is causing this. When I remove it, it works.

I can reproduce the error if I import `bolt11` in components/logger.js.

But why? Why does `bolt11` break the build if imported in components/logger.js?

Reply:
I don't know what problem JS had with logger.js, but moving the code that required `getWalletByName` into wallet-logger.js fixed it. See 4a4d6698.